compiler/vm/compilerbridge.nim

```diff
@@ -491,8 +550,8 @@ proc execProc*(c: var TCtx; sym: PSym; args: openArray[PNode]): PNode =
         kind: rsemWrongNumberOfArguments,
         sym: sym,
         countMismatch: (
-          expected: toInt128(sym.typ.len - 1),
-          got: toInt128(args.len))))
+          expected: sym.typ.len - 1,
+          got: args.len)))
```

Comment:
All the `localReport` stuff to not have `nkError` head out of the Compiler API/nimeval is unnecessary. We don't need to guarantee any API stability for a long time. I'm going to take out all those legacy bits and let the nkError flow through.

Reply:
I will do that in a separate PR, as this one is too big. I was hoping to kill `vmlegacy` entirely in this PR, but that's not in the cards.